examples/tutorials/core_api_pytorch_mnist/model_def_distributed.py

```diff
@@ -234,15 +247,15 @@ def main(core_context):
     optimizer = optim.Adadelta(model.parameters(), lr=hparams["learning_rate"])
     scheduler = StepLR(optimizer, step_size=1, gamma=args.gamma)
 
-    starting_epoch = 0
-    epoch = starting_epoch
+    epoch_idx = epochs_completed
     last_checkpoint_batch = None
 
     for op in core_context.searcher.operations():
 
-        while epoch < op.length:
-            steps_completed = epoch * len(train_loader)
-            train(args, model, device, train_loader, optimizer, core_context, epoch, op)
+        while epoch_idx < op.length:
+            train(args, model, device, train_loader, optimizer, core_context, epoch_idx, op)
```

Comment:
I noticed looking at the proposed docs that there's still an off-by-one error here: you are calling `train(epoch=epoch_idx)` basically but you are calculating steps_completed inside of `train()` based on `(epoch - 1)`, which will dip down to `-1` at `epoch_idx=0`.

I suggest finding all instances of `epoch` in your examples and being consistent as to what should be `epoch_idx` and what should be `epochs_completed`.  In the case of `train()` for instance, I think passing `epoch_idx` makes the most sense, and since `epochs_completed` isn't really "completed" until after `train()` is actually done.

Reply:
resolved